### PR TITLE
Libnetwork Integration with Plugin and Remote Driver Backend

### PR DESCRIPTION
**Pull request by mavenugo**
Now that the modular plugin infra is available as a Docker Package, we can integrate it to support Remote Driver.

This PR integrates Libnetwork and the Plugin Infra to dynamically discover `NetworkDriver`.
Being a modular package, there is no dependancy on the Docker runtime itself as evidenced in the tests added to libnetwork_test.

### CODE REVIEW

File: libnetwork_test.go [hunk 1]
```diff
@@ -1132,6 +1119,102 @@ func TestResolvConf(t *testing.T) {
...
+		t.Fatal("Failed to start a HTTP Server")
+	}
+	defer server.Close()
+
+	type pluginRequest struct {
+		name string
+	}
+
+	mux.HandleFunc("/Plugin.Activate", func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "appplication/vnd.docker.plugins.v1+json")
```
Comment (squaremo) [inline libnetwork_test.go]: typo "appplication"

Comment (mavenugo) [inline libnetwork_test.go]: Thanks. Its a copy+paste issue originating from the plugins code :-(
This needs a fix in the plugins pkg (which we have imported as dependency).

I will first fix the Docker pkg, import it and then change it here to be consistent.

**Comment by mavenugo** [comment]
@squaremo took care of your comments. it certainly simplifies and makes the remote driver more readable.

**Comment by squaremo** [comment]
> took care of your comments

Nice one. The implementation of both lazy- and auto-registration looks good to me.

**Comment by mavenugo** [comment]
Thanks @squaremo . ping @mrjana & @aboch

File: drivers/remote/driver.go [hunk 2]
```diff
@@ -10,13 +12,22 @@ import (
...
 type driver struct {
+	endpoint    *plugins.Client
+	networkType string
 }
 
 // Init does the necessary work to register remote drivers
 func Init(dc driverapi.DriverCallback) error {
+	plugins.Handle(driverapi.NetworkPluginEndpointType, func(name string, client *plugins.Client) {
+
+		// TODO : Handhake with the Remote Plugin goes here
```
Comment (squaremo) [inline drivers/remote/driver.go]: There's no handshake to be done; the plugins package does the handshake when the plugin is loaded, and the `plugins.Handle` hook is called subsequently.

Comment (squaremo) [inline drivers/remote/driver.go]: I'll be touching this anyway, so OK to leave as-is here.

**Comment by mrjana** [comment]
LGTM

**Comment by aboch** [comment]
LGTM

**Pull request merged**